src/kinetics/include/antioch/troe_falloff.h

```diff
@@ -196,40 +196,34 @@ namespace Antioch
   }
 
   template<typename CoeffType>
-  template<typename StateType, typename VectorStateType>
+  template<typename StateType>
   inline
   StateType TroeFalloff<CoeffType>::operator()(const StateType& T,
-                                               const VectorStateType &molar_densities,
+                                               const StateType &M,
                                                const StateType &k0, 
                                                const StateType &kinf) const
   {
 
     //compute log(Fcent) once
     StateType logFcent = ant_log(this->Fcent(T));
-    //[M] 
-    StateType M = Antioch::zero_clone(T);
-    for(unsigned int i = 0; i < molar_densities.size(); i++)
-    {
-        M += molar_densities[i];
-    }
 
     // Pr = [M] * k0/kinf
     ANTIOCH_AUTO(StateType) Pr = M * k0/kinf;
     // c = -0.4 - 0.67 * log10(Fcent)
     // Note log10(x) = (1.0/log(10))*log(x)
-    StateType  c = - 0.4 - _c_coeff*logFcent;
+    StateType  c = - CoeffType(0.4L) - _c_coeff * logFcent;
 
     // n = 0.75 - 1.27 * log10(Fcent)
     // Note log10(x) = (1.0/log(10))*log(x)
-    ANTIOCH_AUTO(StateType) n = 0.75 - _n_coeff*logFcent;
-    ANTIOCH_AUTO(StateType) d = Antioch::constant_clone(T, 0.14);
+    ANTIOCH_AUTO(StateType) n = CoeffType(0.75L) - _n_coeff * logFcent;
+    ANTIOCH_AUTO(StateType) d = constant_clone(T,0.14L);
```

Comment:
Need to wrap this 0.14L in a CoeffType() conversion.

Alternatively, I could add the downconversion in our utilities/ implementations.  I kind of like being alerted to downconversions, though, so we don't make them by accident.